### PR TITLE
Issue1: User can add a trip

### PR DESCRIPTION
**Pull request by jennydtran**
Screen Shot 2020-12-29 at 4 29 19 PM
Screen Shot 2020-12-29 at 4 31 19 PM

Dec-29-2020 16-41-36

### CODE REVIEW

File: scripts/db-import.bash [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash -e
 
-test -f .env && source .env
+source .env
```
Comment (thebearingedge) [inline scripts/db-import.bash:3]: y tho

File: server/public/styles.css [hunk 2]
```diff
@@ -0,0 +1,173 @@
+.hidden {
+  display: none !important;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:3]: Bootstrap has a `d-none` class you can use.
https://getbootstrap.com/docs/4.5/utilities/display/

File: server/public/styles.css [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+* {
+  box-sizing: border-box;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:7]: Boostrap already does this.

File: server/public/styles.css [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+html, body {
+  min-height: 680px;
+  max-width: 900px;
+  margin: 0;
+  padding: 0;
+  font-family: 'Open Sans', Helvetica, Arial, sans-serif;
+  font-size: 16px;
```
Comment (thebearingedge) [inline server/public/styles.css:15]: 16px is the default.

File: server/public/styles.css [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+* {
+  box-sizing: border-box;
+}
+
+html, body {
+  min-height: 680px;
+  max-width: 900px;
+  margin: 0;
+  padding: 0;
```
Comment (thebearingedge) [inline server/public/styles.css:13]: Bootstrap has spacing helpers you should use for margin and padding in most situations.

https://getbootstrap.com/docs/4.5/utilities/spacing/

File: server/public/styles.css [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+main {
+  height: 86.5vh;
+  padding: 0.5rem 0.75rem;
+}
+
+h2 {
+  font-family: 'Signika', Helvetica, sans-serif;
+  font-weight: 700;
+  text-align: center;
```
Comment (thebearingedge) [inline server/public/styles.css:26]: Bootstrap has text helpers you should use in most situations.

https://getbootstrap.com/docs/4.5/utilities/text/

File: server/public/styles.css [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+span {
+  color: rgb(182, 182, 182);
+  font-size: 0.8rem;
+}
+
+
+hr {
+  border: 0;
```
Comment (thebearingedge) [inline server/public/styles.css:44]: Bootstrap has border helpers
https://getbootstrap.com/docs/4.5/utilities/borders/

File: server/public/styles.css [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  background-color: #40B4A6;
+  color: white;
+  border: none;
+  padding: 0.75rem 1.5rem;
+  filter: drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.75));
+}
+
+.text-center {
+  color:#6BDCCE;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:63]: wat

File: server/public/styles.css [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+.homeWelcome > p:nth-child(2) {
+  padding: 0;
+  margin-top: -0.5rem;
+}
+
+.nav {
+  max-width: 900px;
+  background-color: rgb(243, 243, 243);
+  height: 7vh;
```
Comment (thebearingedge) [inline server/public/styles.css:77]: setting height to anything other than px is kinda sketchy. Try a px value

File: server/public/styles.css [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+.icon {
+  color: black;
+  filter: drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.75));
+}
+
+.relative {
+  position: relative;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:119]: Bootstrap has position helpers
https://getbootstrap.com/docs/4.5/utilities/position/

File: server/public/styles.css [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+.overlay {
+  height: 100%;
+  width:100%;
+  position: fixed;
+  z-index: 10;
```
Comment (thebearingedge) [inline server/public/styles.css:125]: Why 10?

File: server/public/styles.css [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+  left: 0;
+  top: 0;
+  background-color: rgba(0,0,0, 0.7);
+  transition: 0.5s;
+  display: flex;
+  justify-content: center;
+  align-items: center;
```
Comment (thebearingedge) [inline server/public/styles.css:132]: Bootstrap has flex helpers.
https://getbootstrap.com/docs/4.5/utilities/flex/

File: server/public/styles.css [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+.trip-modal {
+  width: 85%;
```
Comment (thebearingedge) [inline server/public/styles.css:136]: Instead of a percentage width, would a padding on the parent work?

File: server/index.js [hunk 14]
```diff
@@ -1,10 +1,110 @@
...
+    select *
+      from "trip"
+  `;
+  db.query(sql)
+    .then(result => res.json(result.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/trip', (req, res, next) => {
+  // const { userId } = req.user;
```
Comment (thebearingedge) [inline server/index.js:88]: Please remove commented code.

File: server/index.js [hunk 15]
```diff
@@ -1,10 +1,110 @@
...
+          }
+          const payload = { userId, username };
+          const token = jwt.sign(payload, process.env.TOKEN_SECRET);
+          res.json({ token, user: payload });
+        });
+    })
+    .catch(err => next(err));
+});
+
+// app.use(authorizationMiddleware);
```
Comment (thebearingedge) [inline server/index.js:75]: Please remove commented code.

File: server/index.js [hunk 16]
```diff
@@ -1,10 +1,110 @@
...
+const errorMiddleware = require('./error-middleware');
+const authorizationMiddleware = require('./authorization-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL
+});
 
 const app = express();
 
-app.use(staticMiddleware);
```
Comment (thebearingedge) [inline server/index.js:7]: You definitely should leave this in. It's how the app serves static files.

File: server/authorization-middleware.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
```
Comment (thebearingedge) [inline server/authorization-middleware.js:2]: You can remove these comments, they aren't relevant anymore.

File: client/pages/home.jsx [hunk 18]
```diff
@@ -1,8 +1,206 @@
...
+  }
+}
+
+// function TripEntries(props) {
+//   return (
+//     <div className="container">
+//       <h1>Your Trips</h1>
+//     </div>
+//   );
+// }
```
Comment (thebearingedge) [inline client/pages/home.jsx:206]: ## 👮‍♀️

File: server/public/index.html [hunk 19]
```diff
@@ -6,6 +6,11 @@
   <title>Final Project</title>
   <link rel="icon" href="/favicon.ico">
   <link rel="stylesheet" href="/styles.css">
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css"
```
Comment (thebearingedge) [inline server/public/index.html:9]: I recommend putting your own styles _after_ external styles, just to be sure your own take priority.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Overall looks awesome, but the code can use some clean up. See how much of your custom styles you can trim back.

File: client/pages/home.jsx (hunk 18 above)
Comment (jennydtran) [reply client/pages/home.jsx:206]: Had to stop myself. Realized I was going into the next feature.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**